web3/utils/abi.py

```diff
@@ -1,23 +1,42 @@
+from collections import (
+    namedtuple,
+)
+
 import re
 import itertools
 
+from cytoolz import (
+    curry,
+    pipe,
+)
+
 from eth_utils import (
+    add_0x_prefix,
     coerce_args_to_bytes,
     coerce_args_to_text,
     coerce_return_to_text,
-    to_tuple,
-    add_0x_prefix,
+    is_address,
+    is_boolean,
+    is_integer,
     is_list_like,
     is_string,
-    is_integer,
-    is_boolean,
-    is_address,
+    to_checksum_address,
+    to_tuple,
 )
 
 from eth_abi.abi import (
     process_type,
 )
 
+from web3.utils.formatters import (
+    recursive_map
+)
+
+BASE_RETURN_NORMALIZERS = [
+    lambda typ, data: (typ, to_checksum_address(data)) if typ == 'address' else (typ, data),
```

Comment:
use of a lambda here is bugging me.  lambda + long line + turnary-statement = 😢

Reply:
The simplest alternative is to define a new function just above, but it bugs me to have the constant declared beneath other functions. Which leads me to creating a new module to hold the new formatter, but the name formatters is already taken, and doesn't mean the same thing. I guess web3/util/normalizers.py unless you have other suggestions, then I'll do something like:

```
from web3.util.normalizers import (
  address_to_checksum,
)

BASE_RETURN_NORMALIZERS = [
  address_to_checksum,
]
```

Reply:
I'd be fine with the function definition above.  There's plenty of precedence of that format in my other code....  but you're call.